app/Models/RH/Equipement.php

```
@@ -56,7 +56,9 @@ public function assignments(): HasMany
 
     public function currentAssignment(): HasOne
     {
-        return $this->hasOne(EquipementAssignment::class)->whereNull('returned_at');
+        $relation = $this->hasOne(EquipementAssignment::class);
+
+        return $relation->whereNull('returned_at');
```

PHPStan (2.2.7)

[error] Method App\Models\RH\Equipement::currentAssignment() should return Illuminate\Database\Eloquent\Relations\HasOne but returns Illuminate\Database\Query\Builder.


(return.type)